src/applications/letters/tests/actions/letters.unit.spec.js

```diff
@@ -261,10 +261,16 @@ describe('getLetterListAndBSLOptions', () => {
     const thunk = getLetterListAndBSLOptions();
     const dispatch = () => {};
 
-    thunk(dispatch).then(() => {
-      expect(global.fetch.callCount).to.equal(1);
-      done();
-    });
+    thunk(dispatch)
+      .then(() => {
+        done(
+          new Error('getLetterList should have rejected but resolved instead'),
+        );
+      })
+      .catch(() => {
+        expect(global.fetch.callCount).to.equal(1);
```

Comment:
Can you explain this to me? I think I'm not following. Why does removing the Sentry line in the `getLetterList.catch()` change how `getLetterListAndBSLOptions` gets resolved or rejected?

Reply:
I was getting an unhandled promise rejection from Mocha.

I'm a little confused as to why the old test was passing. If fetch is mocked to return a `Promise.reject()`, why is going to a `.then()` instead of `.catch()`? Changing it to an empty `.catch()` then makes the `thunk(dispatch)` handle an error. Not sure why. Any ideas on what the right way to test this should be?

Reply:
Try the following:
```javascript
Promise.reject()
  .catch(() => console.log('Caught on the first try'))
  .catch(() => console.log('Caught on the second try'));

Promise.reject()
  .catch()
  .catch(() => console.log('The first catch was empty'));

Promise.reject()
  .catch(() => {})
  .catch(() => console.log('Never had a chance'));
```

The empty catch acts as a passthrough. What you can do, though, is add an empty function so that `.catch()` _does_ catch the rejection, but then summarily discards it. If that's what you're looking to do.

Reply:
aha, the MDN page explains that exact scenario at the top: 

> The catch() method returns a Promise and deals with rejected cases only. It behaves the same as calling Promise.prototype.then(undefined, onRejected) (in fact, calling obj.catch(onRejected) internally calls obj.then(undefined, onRejected)). This means that you have to provide onRejected function even if you want to fallback to undefined result value - for example obj.catch(() => {}).

Muchas gracias for the example!